src/NUnitConsole/nunit3-console/ConsoleRunner.cs

```diff
@@ -433,11 +435,51 @@ public static TestPackage MakeTestPackage(ConsoleOptions options)
                 package.AddSetting(FrameworkPackageSettings.DefaultTestNamePattern, options.DefaultTestNamePattern);
 
             if (options.TestParameters != null)
-                package.AddSetting(FrameworkPackageSettings.TestParameters, options.TestParameters);
+            {
+                var dictionary = new Dictionary<string, string>();
+
+                // Exactly what the framework used to do.
+                // Now can be independently changed without breaking backwards compatibility with old frameworks.
+                foreach (var param in options.TestParameters.Split(';'))
```

Comment:
> Use a review comment (like this one) if you want to tell the reviewers why you did something.

Thanks for the suggestion. I had this in mind when I did it, but future bug fix es will involve seeing the code comments and will not involve seeing these comments. It was important to me that those commends stay with the code.